x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -467,10 +473,18 @@ public void testMaxRestoreBytesPerSecIsUsed() throws Exception {
             // which is defined in RateLimitingInputStream (see details in https://github.com/elastic/elasticsearch/pull/96444)
             long minimax = getMaxShardSizeByNodeInBytes(indexName).values().stream().min(Long::compareTo).get();
             long rateLimitInBytes = (long) (0.9 * minimax);
-            repositorySettings.put("max_restore_bytes_per_sec", ByteSizeValue.ofBytes(rateLimitInBytes));
+            repositorySettings.put(MAX_RESTORE_BYTES_PER_SEC.getKey(), ByteSizeValue.ofBytes(rateLimitInBytes));
         } else {
-            repositorySettings.put("max_restore_bytes_per_sec", ByteSizeValue.ZERO);
+            repositorySettings.put(MAX_RESTORE_BYTES_PER_SEC.getKey(), ByteSizeValue.ZERO);
+        }
+
+        if (useRateLimits == false) {
+            // Disable rate limiter which is defined in RecoverySettings
+            // BlobStoreRepository#maybeRateLimitRestores uses two rate limiters and a single callback listener
+            // which is asserted in this test. Both rate limiter should be switched off
```

Comment:
you can simply put the `updateClusterSettings` in the else branch of the condition above.

nit `limiters`